services/datamanager/builtin/capture_test.go

```diff
@@ -333,6 +359,153 @@ func getSensorData(dir string) ([]*v1.SensorData, error) {
 	return sd, nil
 }
 
+func TestCollector(t *testing.T) {
```

Comment:
If the goal here is to test the various component collectors produce correctly formatted data, is there a need to use the datamanager service as a pass through like this? What's the advantage vs directly building the collectors with the `newXCollector` constructors in each component/collectors.go file?

Reply:
There isn't a particular need with respect to the goal. I did it here to be able to test all the collectors in the same place instead of having it per collector file and keeping all the configs in the same place. I can change it if thats better.

Reply:
Since the data manager is only being used as a sort of indirect passthrough to call start/stop on the underlying collector, I think it might obfuscate what we're actually testing more than it helps from being a bit more integration-y. So I would personally err towards putting it in some collector_test.go file in the component directories. I think this would allow you to not need the config files too, since you can just directly use the CollectorParams struct to build them. Very open to other opinions and arguments though (cc @agreenb @dmhilly)

Reply:
Yeah I agree with you Aaron that that would be simpler. I may have put this idea in Vijay's head since I had thought the tests might work by running data capture and also making a grpc call to the same method and comparing the responses to make sure the format is the same. But I think comparing with the known grpc response message is doing pretty much the same thing and is simpler, so I'd be in favor of changing this to take out the datamanager layer, just constructing the component and passing it into the collector and testing that.

Reply:
Makes sense, will do